pkg/scheduler/objects/events/application_events.go

```diff
@@ -79,6 +79,14 @@ func (ae *ApplicationEvents) SendRemoveAllocationEvent(appID, allocKey string, a
 	ae.eventSystem.AddEvent(event)
 }
 
+func (ae *ApplicationEvents) SendPreemptAllocationEvent(appID, allocKey string, allocated *resources.Resource, message string) {
+	if !ae.eventSystem.IsEventTrackingEnabled() {
+		return
+	}
+	event := events.CreateAppEventRecord(appID, message, allocKey, si.EventRecord_REMOVE, si.EventRecord_ALLOC_PREEMPT, allocated)
```

Comment:
This must be a `REQUEST` type. That will result in a real K8s event being sent to the target pod.

The removal of the allocation will be a natural consequence of that, which will flow through the scheduler, so an app-specific event is not needed here.